src/pkg/clients/Service_discovery.go

```diff
@@ -36,49 +36,70 @@ func NewServiceDiscovery(host, user, password string) (ServiceDiscovery, error)
+		agent:                     client.Agent(),
+		transformersAddressesList: map[string][]string{},
 	}
-	return s.parseTransformerInfos(services), nil
-}
 
-// Get all available instances of a given transformation service
-func (s serviceDiscovery) GetTransformationServicesWithName(name string) ([]*TransformerInfos, error) {
-	services, err := s.agent.ServicesWithFilter(name + " in Service and transformer in Service")
-	if err != nil {
-		return nil, err
-	}
-	return s.parseTransformerInfos(services), nil
+	return &service, nil
 }
 
-func (s serviceDiscovery) parseTransformerInfos(services map[string]*consul_api.AgentService) []*TransformerInfos {
-	transformers := make([]*TransformerInfos, 0, len(services))
-	for _, service := range services {
-		transformers = append(transformers, &TransformerInfos{
-			Name:    service.Service,
-			Address: service.Address,
-			Port:    strconv.Itoa(service.Port),
-		})
+// Get all available instances of a given transformation service
+func (s *serviceDiscovery) GetTransformationServices(name string) ([]string, error) {
+	if len(s.transformersAddressesList[name]) < 1 {
+		return nil, fmt.Errorf("No service with name %v found.", name)
 	}
-	return transformers
+	return s.transformersAddressesList[name], nil
 }
 
 // Register service, will be use only for local dev
-func (s serviceDiscovery) RegisterService(name, address string, port int, tags []string) error {
+func (s *serviceDiscovery) RegisterService(name, address string, port int, tags []string) error {
 	return s.agent.ServiceRegister(&consul_api.AgentServiceRegistration{
 		Name:    name,
 		Address: address,
 		Port:    port,
 		Tags:    tags,
 	})
 }
+
+// Get all available instances of transformation services
+func (s *serviceDiscovery) updateList() error {
+	services, err := s.agent.ServicesWithFilter("transformer in Service")
+	if err != nil {
+		return err
+	}
+	s.parseTransformerList(services)
+	return nil
+}
+
+func (s *serviceDiscovery) parseTransformerList(services map[string]*consul_api.AgentService) {
+	s.transformersAddressesList = map[string][]string{}
+	for _, service := range services {
+		name := strings.Split(service.Service, "-")[0]
+		address := service.Address + ":" + strconv.Itoa(service.Port)
+		s.transformersAddressesList[name] = append(s.transformersAddressesList[name], address)
+	}
+}
+
+func (s *serviceDiscovery) Watch() error {
+	// Create Watch that check for changes on services register/deregister
+	plan, err := watch.Parse(map[string]interface{}{"type": "services"})
+	if err != nil {
+		return err
+	}
+
+	// Define the handler function that will be called for each change
+	plan.Handler = func(idx uint64, result interface{}) {
+		log.Info("Change detected : Service register/deregister")
+		log.Debug("index = ", idx, "\n", "result=", result)
+		_ = s.updateList()
+	}
+
+	// Launch the watch. Note that the handler function will be run one first time
+	err = plan.RunWithClientAndHclog(s.client, nil)
```

Comment:
OK, here stoping the watcher seem to work with client and not context https://github.com/hashicorp/consul/blob/api/v1.13.1/api/watch/plan.go#L59